docs/src/dev/use-cases.md

```diff
@@ -0,0 +1,54 @@
+
+**Proving Block Execution**
+
+Ziren is used to generate ZK proofs for individual blocks processed by GOAT’s sequencers. Each proof demonstrates that a block was executed correctly and that its resulting state transitions are valid. To amortize costs, proofs are generated over a specific period rather than for each transaction individually.
+
+The proof contains the hash of the associated block, root hash of the sequencer set and new state root. These values are bundled and submitted to the Bitcoin L1. The block hash and the root hash serve as public inputs to the ZK circuit and are required to match the corresponding values that have already been posted on Bitcoin L1.
+
+**Real-Time Proving & Peg-Out Proofs**
+
+Ziren enables the generation of proofs in real time for every block on GOAT, which powers the BitVM2 bridge used for withdrawals during the peg-out process. This capability eliminates the delays that can otherwise occur during withdrawals, allowing funds to be released in sync with block production rather than requiring days (or even weeks) of waiting.
+
+During the peg-out process, operators act as provers to generate ZK proofs for peg-out transactions. Challengers can contest invalid proofs within an optimistic challenge window.
+
+The proof generation pipeline for peg-outs consists of several stages:
+
+- Root Prover: Computes core execution trace.
+- Aggregation Prover: Compresses multiple core proofs.
+- SNARK Prover: Generates Groth16 proofs (allowing for an EVM-compatible format) for Bitcoin verification.
+
+This architecture enables low-latency withdrawals and the rapid inclusion of new blocks. View proofs being generated in real-time for GOAT, including details for each proof in the pipeline here: https://bitvm2.testnet3.goat.network/proof.
+
+**Proof Verification via BitVM Paradigm**
+
+Once generated, proof commitments are submitted to a BitVM2-anchored contract on Bitcoin L1 and is verified via BitVM2’s fraud-proof challenge mechanism. Verification involves confirming that the public inputs of the ZK proof correspond to a valid transaction on Bitcoin L1 and that this transaction is finalized within a sufficiently long proof-of-work chain.
+
+All ZK proofs are tied to the committed state; any attempt to manipulate the state would produce a different public input, which would cause verification to fail. Both the L2 state and its execution are fully auditable by any participant.
+
+**Transaction Lifecycle**
+
+The following is an overview of the transaction lifecycle on GOAT and Ziren’s role in it:
+
+1. A user first submits a transaction to the L2.
+2. The sequencer batches and executes transactions into an L2 block, which is then processed by Ziren.
+3. Ziren generates a ZK proof that attests to the correctness of the block’s execution and its state transitions.
+4. The ZK proof generated by Ziren, together with the state commitment, are submitted to a BitVM2-anchored contract on the Bitcoin 1
```

Comment:
BitVM2-anchored contract on the Bitcoin 1
=> 
BitVM2-anchored covenant on the Bitcoin.